src/transformStateWithClones.js

```diff
@@ -6,8 +6,42 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(state, transforms) {
```

Comment:
The `object` variable is being modified directly in each operation. This can lead to incorrect results because each transformation should be applied to a fresh copy of the initial state. Consider creating a new copy of the state inside each operation function (addProperties, removeProperties, clearProperties) instead of modifying the same `object` reference.